internal/plugin/grpc_client_plugin_test.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) F5, Inc.
+//
+// This source code is licensed under the Apache License, Version 2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+
+package plugin
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/nginx/agent/v3/internal/bus"
+	"github.com/nginx/agent/v3/internal/config"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func getAgentConfig() *config.Config {
+	return &config.Config{
+		DataPlaneAPI: &config.DataPlaneAPI{
+			Host: "127.0.0.1",
+			Port: 8989,
+		},
+		Command: &config.Command{
+			Server: &config.ServerConfig{
+				Host: "127.0.0.1",
+				Port: 8888,
+				Type: "grpc",
+			},
+		},
+		Client: &config.Client{
+			Timeout: 5 * time.Second,
+		},
+		ProcessMonitor: &config.ProcessMonitor{
+			MonitoringFrequency: time.Millisecond,
+		},
+	}
+}
+
+func TestGrpcClient_NewGrpcClient(t *testing.T) {
+	tests := []struct {
+		name        string
+		agentConfig *config.Config
+		expected    *GrpcClient
+	}{
+		{
+			"grpc config",
+			getAgentConfig(),
+			&GrpcClient{},
+		},
+		{
+			"not grpc type",
+			&config.Config{
+				Command: &config.Command{
+					Server: &config.ServerConfig{
+						Host: "127.0.0.1",
+						Port: 8888,
+						Type: "http",
+					},
+				},
+			},
+			nil,
+		},
+		{
+			"nil client",
+			nil,
+			nil,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
Technically a shadow declaration of the original `t`, weird that the linter didn't pick this up.